wagtailautocomplete/views.py

```diff
@@ -63,20 +64,34 @@ def search(request):
     except ValueError:
         return HttpResponseBadRequest()
 
-    field_name = getattr(model, 'autocomplete_search_field', 'title')
+    field_name = getattr(target_model, 'autocomplete_search_field', 'title')
     filter_kwargs = dict()
     filter_kwargs[field_name + '__icontains'] = search_query
-    queryset = model.objects.filter(**filter_kwargs)
+    queryset = target_model.objects.filter(**filter_kwargs)
 
     if getattr(queryset, 'live', None):
         # Non-Page models like Snippets won't have a live/published status
         # and thus should not be filtered with a call to `live`.
         queryset = queryset.live()
 
-    exclude = request.GET.get('exclude', '')
-    if exclude:
-        exclusions = [unquote(item) for item in exclude.split(',') if item]
-        queryset = queryset.exclude(pk__in=exclusions)
+    # Get exclusions based on the existing instance.
+    instance = request.GET.get('instance', '')
+    db_field = request.GET.get('db_field', '')
+    if instance and db_field:
+        model_str, field_str = db_field.rsplit(".", 1)
+
+        try:
+            model = apps.get_model(model_str)
+        except Exception:
+            return HttpResponseBadRequest()
+
+        if isinstance(model._meta.get_field(field_str), ManyToManyField):
+            model_instance = model.objects.get(pk=instance)
+            model_field = getattr(model_instance, field_str)
+            exclusions = model_field.all().values_list("pk", flat=True).distinct()
+            queryset = queryset.exclude(pk__in=exclusions)
+        else:
+            queryset = queryset.exclude(pk=instance)
```

Comment:
I don't think this is the correct query? shouldn't this do something with the db_field?

Reply:
Yeah you were right, I've fixed the logic here.